fix: InstalledAppInfoStatus type

matthme opened the pull request:
### Summary
- fixes the `InstalledAppInfoStatus` type.

### TODO:
- [x] CHANGELOG mentions all code changes.
- [x] docs have been updated (`npm run build:docs`)

matthme committed:
fixed InstalledAppInfoStatus type

mattyg reviewed (commented):
I think `PausedAppReason` and `DisabledAppReason` need to be updated as well, right?

Also I see there is a type `DeactivationReason` that I can't find used anywhere - can it be removed?

matthme committed:
fix more types, remove unused type

matthme commented:
You catch my sloppiness in bunches. Thanks! I updated the other types as well, removed `DeactivationReason` and also renamed `InstalledAppInfoStatus` to `AppInfoStatus` as it's being called on the holochain side for a while now.

matthme committed:
udpate changelog
